pkg/opensearch/snapshot_tests/README.md

```diff
@@ -0,0 +1,35 @@
+## Contributing to snapshot tests
+
+#### Tests
+* File names begin with query_type, e.g. `lucene_raw_data_test.go`
+* Each file/query type contains two tests:
+  * Test_query_type_request: Given query input from the data source frontend in Grafana, validates the request made to OpenSearch
+  * Test_query_type_response: Given the response from OpenSearch, validates the snapshot of a data frame generated by the data source at the time of writing the test
+
+#### `testdata` file names
+* Begins with the query type, e.g. `lucene_raw_data`
+* Then the role of the file in the tests: 
+  * `query_input`: from Grafana, the query JSON from the data source frontend to the backend
+  * `response_from_opensearch`
+  * `expected_result_generated.golden.jsonc` - data frame snapshots - when this assertion fails, some non-regression has not passed
+
+#### To add a request test:
+1. Create a new empty query_input file in `testdata` with the new query type, e.g. `lucene_logs.query_input.json`
+2. Access an instance of Grafana with this data source. 
+3. Execute a query type for which we are adding a new test. 
+4. Use the Network tab to access the payload. Copy only the `queries` array into the `lucene_logs.query_input.json` file.
+5. Up one level in `snapshot_tests`, create a new file `lucene_logs_test.go`
+6. Copy an existing test `Test_query_type_request` and find and replace the query type with your query type e.g. replace instances of `lucene_raw_data` with `lucene_logs` in order to follow the naming conventions. Make sure any references to file names have been searched and replaced.
+7. Run the test and replace the expected request until it passes.
+8. Make sure it makes sense, you could query in OpenSearch with DevTools. 
+
+#### To add a response test:
+1. To intercept the response from OpenSearch, set `debugEnabled = true` on the `baseClientImpl`. We'll use some existing code which reads the response body. There's a [method](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L458) that can be called. 
+2. Access an instance of Grafana with this data source and execute a query type for which we are adding a new test. You may want to limit the response by limiting the time range or setting a smaller size. 
+3. Set a breakpoint right after the body is read [here for Lucene](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L364) and [here for PPL](https://github.com/grafana/opensearch-datasource/blob/b2812efd4b20bafbccf0fa3378b96fb57f2f0b43/pkg/opensearch/client/client.go#L585). The `tmpBytes` JSON is the response from OpenSearch.
+4. Create a new empty response_from_opensearch file in `testdata` with the new query type, e.g.: `lucene_logs.response_from_opensearch.json`
+5. Copy the `tmpBytes` JSON into this file. 
```

Comment:
it was difficult for me to copy the response while debugging with vscode. the response was truncated in the debugger so i ended up just printing it out with `backend.Logger.Debug("tmpBytes", "string(tmpBytes)", string(tmpBytes))` and then running it through `JSON.parse` with javascript.

am i missing an easier way to copy this? or is it just difficult with vscode?

Reply:
Darn! I was able to copy it with my Goland. Maybe it wasn't as long. Do you want me to try with VSCode?
Another idea is maybe you can query the OpenSearch dashboard directly? 
Did limiting the time range help or was setting a smaller size possible?

Reply:
i can only limit it to 3 fields and a single row.

i think there's some character limit in the debugger or some other setting that limits it.